src/C4EnergyBars.h

```diff
@@ -0,0 +1,263 @@
+/*
+ * LegacyClonk
+ *
+ * Copyright (c) RedWolf Design
+ * Copyright (c) 2017-2019, The LegacyClonk Team and contributors
+ *
+ * Distributed under the terms of the ISC license; see accompanying file
+ * "COPYING" for details.
+ *
+ * "Clonk" is a registered trademark of Matthes Bender, used with permission.
+ * See accompanying file "TRADEMARK" for details.
+ *
+ * To redistribute this file separately, substitute the full license texts
+ * for the above references.
+ */
+
+/* Allows to define custom energy bars */
+
+#pragma once
+
+class C4EnergyBars;
```

Comment:
Oh, I selected too much. I meant only the forward declaration.

Reply:
Without it I get:
```
/home/sze/games/LegacyClonk/src/C4Object.h:202:25: error: ‘C4EnergyBars’ was not declared in this scope
  202 |         std::shared_ptr<C4EnergyBars> energyBars;
      |                         ^~~~~~~~~~~~
/home/sze/games/LegacyClonk/src/C4Object.h:202:37: error: template argument 1 is invalid
  202 |         std::shared_ptr<C4EnergyBars> energyBars;
      |                                     ^
```
Because C4Object needs C4EnergyBars and C4EnergyBars needs reference to C4Object in its constructor.
I am not sure how to solve it without the forward declaration.